renpy/revertable.py

```diff
@@ -335,11 +336,15 @@ def has_key(self, key):
 
         # https://peps.python.org/pep-0584 methods
         def __or__(self, other):
+            if not isinstance(other, dict):
+                return NotImplemented
             rv = RevertableDict(self)
             rv.update(other)
             return rv
 
         def __ror__(self, other):
+            if not isinstance(other, dict):
```

Comment:
Specifically for `__ror__`, I think that this should be either a direct type or isinstance check of RevertableDict rather than dict.

The impact of such a change would be near non-existant `rpy` files as the RHS `dict` or `{}` would be implicitly RevertableDicts already and so `__or__` would be used, however in `py` files, the distinction between `dict() | RevertableDict() -> dict` and `RevertableDict() | dict() -> RevertableDict` would be useful.

Reply:
You would have `if type(other) is not dict: return Notimplemented` ? I understand the context but I'm unsure about what you'd want to see.

Reply:
I don't recall if we support (or want to) subclassing of Revertable types, if we do then I'd suggest:
```suggestion
            if not isinstance(other, RevertableDict):
```
but if we're not expecting (or wanting) subtypes then as you noted in the list change, we could direct check:
```suggestion
            if type(other) is not RevertableDict:
```